src/helper/html.rs

```diff
@@ -186,176 +180,98 @@ pub struct HfdElement {
     classes: Vec<String>,
 }
 impl HfdElement {
-    pub fn get_by_name(&self, name: &str) -> bool {
-        let mut result = false;
-        if self.name == name {
-            result = true;
-        }
-        result
+    pub fn has_name(&self, name: &str) -> bool {
+        self.name == name
     }
 
     pub fn get_by_name_and_attribute(&self, name: &str, attribute: &str) -> Option<String> {
-        let mut result: Option<String> = None;
-        if self.name == name {
-            match self.attributes.get(attribute) {
-                Some(v) => {
-                    result = Some(v.to_string());
-                }
-                _ => {}
-            }
-        }
-        result
+        self.attributes
+            .get(attribute)
+            .and_then(|v| (self.name == name).then(|| v.to_string()))
     }
 
-    pub fn get_by_class(&self, class: &str) -> bool {
-        let mut result = false;
-        for cls in &self.classes {
-            if class == cls {
-                result = true;
-            }
-        }
-        result
+    pub fn contains_class(&self, class: &str) -> bool {
+        self.classes.contains(&class.to_string())
     }
 }
 
 pub trait AnalysisMethod {
+    fn is_tag(&self, html_flat_data: &HtmlFlatData, tag: &str) -> bool;
     fn font_name(&self, html_flat_data: &HtmlFlatData) -> Option<String>;
     fn size(&self, html_flat_data: &HtmlFlatData) -> Option<f64>;
     fn color(&self, html_flat_data: &HtmlFlatData) -> Option<String>;
-    fn bold(&self, html_flat_data: &HtmlFlatData) -> bool;
-    fn italic(&self, html_flat_data: &HtmlFlatData) -> bool;
-    fn underline(&self, html_flat_data: &HtmlFlatData) -> bool;
-    fn superscript(&self, html_flat_data: &HtmlFlatData) -> bool;
-    fn subscript(&self, html_flat_data: &HtmlFlatData) -> bool;
-    fn strikethrough(&self, html_flat_data: &HtmlFlatData) -> bool;
+    fn is_bold(&self, html_flat_data: &HtmlFlatData) -> bool;
+    fn is_italic(&self, html_flat_data: &HtmlFlatData) -> bool;
+    fn is_underline(&self, html_flat_data: &HtmlFlatData) -> bool;
+    fn is_superscript(&self, html_flat_data: &HtmlFlatData) -> bool;
+    fn is_subscript(&self, html_flat_data: &HtmlFlatData) -> bool;
+    fn is_strikethrough(&self, html_flat_data: &HtmlFlatData) -> bool;
```

Comment:
The trait function names changed to describe the functionality and potentially a **breaking** change.